### PR TITLE
Patch/border box size and safari mobile bug

### PR DESCRIPTION
**Pull request by aetheon**
https://digitalcrew.teamwork.com/#/tasks/10961385

### CODE REVIEW

File: dist/autosize.js [hunk 1]
```diff
@@ -77,6 +77,8 @@
 
 			if (style.boxSizing === 'content-box') {
 				heightOffset = -(parseFloat(style.paddingTop) + parseFloat(style.paddingBottom));
+			} else if (style.boxSizing === 'border-box') {
```
Comment (aetheon) [inline dist/autosize.js:80]: This is a fix for:
1. Add several lines to textarea
2. Delete all lines 

Then the text-area will be a few pixels taller than it should be

File: dist/autosize.js [hunk 2]
```diff
@@ -109,10 +111,14 @@
 			var arr = [];
 
 			while (el && el.parentNode && el.parentNode instanceof Element) {
-				if (el.parentNode.scrollTop) {
+				var scrollTop = el.parentNode.scrollTop;
+				if (el.parentNode === document.body) {
+					scrollTop = document.documentElement.scrollTop;
```
Comment (aetheon) [inline dist/autosize.js:116]: this is the fix for the mobile problem. Basically body.scrollTop doesn't work as expected in some browsers, like FF. My best guest is that the latest webkit on safari is using something similar.

The library was already using something like it, but they forgot this.

File: src/autosize.js [hunk 3]
```diff
@@ -87,11 +89,15 @@ function assign(ta) {
 		const arr = [];
 
 		while (el && el.parentNode && el.parentNode instanceof Element) {
-			if (el.parentNode.scrollTop) {
+			var scrollTop = el.parentNode.scrollTop;
+			if(el.parentNode === document.body){
+				scrollTop = document.documentElement.scrollTop;
+			}
+			if (scrollTop) {
```
Comment (mike182uk) [inline src/autosize.js:96]: Is there any need for this check? From what i can see `scrollTop` is always going to have an assignment? (either `el.parentNode.scrollTop` or `document.documentElement.scrollTop`)?

File: src/autosize.js [hunk 4]
```diff
@@ -55,6 +55,8 @@ function assign(ta) {
 
 		if (style.boxSizing === 'content-box') {
 			heightOffset = -(parseFloat(style.paddingTop)+parseFloat(style.paddingBottom));
+		} else if (style.boxSizing === 'border-box') {
+			heightOffset = -(parseFloat(style.borderTopWidth) + parseFloat(style.borderBottomWidth));
```
Comment (mike182uk) [inline src/autosize.js]: Maybe remove the spacing around the `+` to be consistent with the other statements

File: src/autosize.js (hunk 3 above)
Comment (aetheon) [reply src/autosize.js:96]: basically what this is doing is setting it if different than zero. Same behaviour as before, don't wanna change it 😩

Comment (mike182uk) [reply src/autosize.js:96]: But doesn't your change make the `if` statement redundant as `scrollTop` will always have an assignment (either `el.parentNode.scrollTop` or `document.documentElement.scrollTop`)

Comment (aetheon) [reply src/autosize.js:96]: > scrollTop will always have an assignment

That was using node.scrollTop. This cannot be used on the body element, therefore the fix where:
```
scrollTop = document.documentElement.scrollTop;

instead of 

scrollTop = body.scrollTop;
```

using this the value will be zero on safari. you get me?

Comment (mike182uk) [reply src/autosize.js:96]: Yep i get you, i don't think you get me though 😛

I was trying to assess whether you still needed the `if` statement. On a second glance, the logic in the `if` will not be executed if `el.parentNode.scrollTop` is falsy, and this behaviour is probably still needed as `el.parentNode.scrollTop` can be still be falsy if `el.parentNode !== document.body`.

**TLDR;**

It's fine.

**Review by mike182uk — APPROVED** [review]

File: src/autosize.js (hunk 3 above)
Comment (aetheon) [reply src/autosize.js:96]: but do you still think we need it? I can't see any code path that actually needs that. don't give up on me lol
We'll chat tomorrow, just to make sure.
